### PR TITLE
#52 feat: unmount and mount

### PR DESCRIPTION
**Pull request by trntuan**
+ unmound categories
+ unmound banner

### CODE REVIEW

File: src/redux/category/Reducer.ts [hunk 1]
```diff
@@ -34,6 +34,12 @@ const CategoriesReducer = (state: InitialInterface = initialState, action: Categ
 				error: payload,
 				isLoading: false
 			};
+		// Todo: 2. Success
+		case TYPES.SET_CATEGORY_UNMOUNT:
+			return {
+				...state,
+				categories: payload,
```
Comment (fdhhhdjd) [inline src/redux/category/Reducer.ts:41]:
```javascript 
   return {
       isLoading: false,
	categories: [],
	error: null
  }

```

File: src/redux/category/Actions.ts [hunk 2]
```diff
@@ -28,6 +28,14 @@ const getDataCategoriesError = (error: Error) => {
 	} as const;
 };
 
+export const setDataCategoriesUnmount = (data: CategoriesItem[]) => {
+	return {
```
Comment (fdhhhdjd) [inline src/redux/category/Actions.ts:32]:
```javascript 
return {
		type: TYPES.SET_CATEGORY_UNMOUNT,
	} as const;

```

File: src/pages/index/categories.tsx [hunk 3]
```diff
@@ -18,6 +18,9 @@ export const CategoriesDev: FC = () => {
 
 	useEffect(() => {
 		dispatch(getCategoriesInitiate());
+		return () => {
+			dispatch(setDataCategoriesUnmount([]));
```
Comment (fdhhhdjd) [inline src/pages/index/categories.tsx:22]: `return () => dispatch(setDataCategoriesUnmount())`

File: src/redux/banner/Reducer.ts [hunk 4]
```diff
@@ -33,6 +33,11 @@ const BannerReducer = (state: InitialInterface = initialState, action: BannerAct
 				error: payload,
 				isLoading: false
 			};
+		case TYPES.SET_BANNER_UNMOUNT:
```
Comment (fdhhhdjd) [inline src/redux/banner/Reducer.ts:36]:
```javascript
return {
		isLoading: false,
		banners: [],
                error: null
};
```

File: src/redux/banner/Actions.ts [hunk 5]
```diff
@@ -28,6 +28,13 @@ const getDataBannerError = (error: Error) => {
 	} as const;
 };
 
+export const setDataBannerUnmount = (data: BannerItem[]) => {
+	return {
+		type: TYPES.SET_BANNER_UNMOUNT,
+		payload: data
```
Comment (fdhhhdjd) [inline src/redux/banner/Actions.ts:34]: remove `payload: data`

File: src/pages/index/banner.tsx [hunk 6]
```diff
@@ -16,6 +16,9 @@ export const Banner: React.FC = () => {
 
 	React.useEffect(() => {
 		dispatch(getBannerInitiate());
+		return () => {
+			dispatch(setDataBannerUnmount([]));
```
Comment (fdhhhdjd) [inline src/pages/index/banner.tsx:20]: `return () => dispatch(setDataBannerUnmount())`

File: src/pages/index/categories.tsx (hunk 3 above)
Comment (trntuan) [reply src/pages/index/categories.tsx:22]: return () =>{
			dispatch(setDataCategoriesUnmount());
		};

**Pull request merged**